### PR TITLE
Add allowed external resolver URLs

### LINKED ISSUE #41: External context id resolution should have URL validation from config
When a notification is received, and the context id is not a DSpace URL we afford to use a HEAD request to resolve. This creates a vulnerability for notifications to be received in which attempt malicious HEAD requests.

Potential server-side request forgery due to [a user-provided value](https://lgtm.com/projects/g/DSpace/DSpace/snapshot/2f5eca7ae2253958947c8fb67398f3545e3f89d5/files/dspace-ldn/src/main/java/org/dspace/app/ldn/processor/LDNMetadataProcessor.java?sort=name&dir=ASC&mode=heatmap#x2c659545261898c3:1).

https://lgtm.com/rules/1513978414265/

To remedy this a configuration of a list of acceptable resolver base URLs should be added and used to validate the external context ids.

### PR DESCRIPTION
**Pull request by wwelling**
Resolves #41 

- add allowed external resolver URLs config `ldn.notify.allowed-external-resolver-urls`
- validate external context ids against allowed external resolver URLs

### CODE REVIEW

File: dspace-ldn/src/main/java/org/dspace/app/ldn/processor/LDNMetadataProcessor.java [hunk 1]
```diff
@@ -94,6 +96,13 @@ private LDNMetadataProcessor() {
 
     @PostConstruct
     public void init() {
+        String allowedExternalResolverUrls = configurationService.getProperty("ldn.notify.allowed-external-resolver-urls");
```
Comment (awoods) [inline dspace-ldn/src/main/java/org/dspace/app/ldn/processor/LDNMetadataProcessor.java:99]: To eliminate the ambiguity, could you name the local String variable, `allowedExternalResolverUrls`, something different than the Array class member, `allowedExternalResolverUrls`?

File: dspace/config/modules/ldn.cfg [hunk 2]
```diff
@@ -15,16 +16,16 @@ ldn.notify.local-inbox-endpoint = ${dspace.server.url}/ldn/inbox
 # Each IDs must match with the ID returned by the external service 
 # in the JSON-LD Actor field
 
-service.service-id.ldn = dev-hdc3b.lib.harvard.edu/api/inbox
+service.service-id.ldn = hdc-dash-api-dev.lib.harvard.edu/api/inbox
```
Comment (awoods) [inline dspace/config/modules/ldn.cfg:19]: Presumably `dev`, `QA`, and `Prod` will all have different values for these variables... managed by our Ansible config. Will that work here?

**Review by awoods — CHANGES_REQUESTED** [review]

File: dspace/config/modules/ldn.cfg (hunk 2 above)
Comment (wwelling) [reply dspace/config/modules/ldn.cfg:19]: Yes, and that is defined by the IF project local.cfg files. I wanted the defaults to be dev for example.

File: dspace-ldn/src/main/java/org/dspace/app/ldn/processor/LDNMetadataProcessor.java (hunk 1 above)
Comment (wwelling) [reply dspace-ldn/src/main/java/org/dspace/app/ldn/processor/LDNMetadataProcessor.java:99]: Yeah, I can do that.

Comment (wwelling) [reply dspace-ldn/src/main/java/org/dspace/app/ldn/processor/LDNMetadataProcessor.java]: Done.

**Review by awoods — APPROVED** [review]

**Pull request merged**